### PR TITLE
Add a support for JSON representation of the trace.

### PR DESCRIPTION
**Pull request by vasilvv**

### CODE REVIEW

File: tools/transform_quic_trace.cc [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+using google::protobuf::util::JsonOptions;
+using google::protobuf::util::JsonStringToMessage;
+using google::protobuf::util::MessageToJsonString;
+
+bool IsValidFormatString(const std::string& format) {
+  return format == "protobuf" || format == "json";
+}
+
+bool InputTrace(quic_trace::Trace* trace) {
+  if (FLAGS_input_format == "json") {
```
Comment (janaiyengar) [inline tools/transform_quic_trace.cc]: drive-by: Suggest if (FLAGS_input_format == "protobuf") instead. Fewer lines need to be indented.

File: tools/transform_quic_trace.cc [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    auto status = JsonStringToMessage(std::string(it, end), trace);
+    return status.ok();
+  }
+
+  return trace->ParseFromIstream(&std::cin);
+}
+
+void OutputTrace(const quic_trace::Trace& trace) {
+  if (FLAGS_output_format == "json") {
```
Comment (janaiyengar) [inline tools/transform_quic_trace.cc]: ditto

File: tools/transform_quic_trace.cc [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    std::istreambuf_iterator<char> end;
+
+    auto status = JsonStringToMessage(std::string(it, end), trace);
+    return status.ok();
+  }
+
+  return trace->ParseFromIstream(&std::cin);
+}
+
+void OutputTrace(const quic_trace::Trace& trace) {
```
Comment (janaiyengar) [inline tools/transform_quic_trace.cc]: perhaps make this a bool, to percolate up whatever MessageToJsonString() returns

File: tools/transform_quic_trace.cc [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+#include <istream>
+
+#include "gflags/gflags.h"
+#include "google/protobuf/util/json_util.h"
+#include "lib/quic_trace.pb.h"
+
+DEFINE_bool(whitespace, true, "Add whitespace to the JSON output");
+DEFINE_string(input_format, "protobuf", "Can be 'protobuf' or 'json'.");
+DEFINE_string(output_format, "protobuf", "Can be 'protobuf' or 'json'.");
+DEFINE_int64(truncate_after_time_us,
```
Comment (ianswett) [inline tools/transform_quic_trace.cc]: You can remove the end, but not the beginning of a trace?  Any reason?

File: tools/transform_quic_trace.cc (hunk 1 above)
Comment (vasilvv) [reply tools/transform_quic_trace.cc]: Done.

File: tools/transform_quic_trace.cc (hunk 2 above)
Comment (vasilvv) [reply tools/transform_quic_trace.cc]: Done.

File: tools/transform_quic_trace.cc (hunk 3 above)
Comment (vasilvv) [reply tools/transform_quic_trace.cc]: I don't think it returns anything?

File: tools/transform_quic_trace.cc (hunk 4 above)
Comment (vasilvv) [reply tools/transform_quic_trace.cc]: You could remove the beginning, but you'd have to recompute the timestamps, which is something I didn't want to write yet.
